File: leetcode/easy/3658-gcd-of-odd-and-even-sums/solution.py
class Solution:
    def gcdOfOddEvenSums(self, n: int) -> int:
        esum,osum=0,0

        for i in range(1,2*n+1):
            if i%2 != 0:
                osum += i
            else:
                esum += i
        
        while(esum!=0):
            temp = esum
            esum = osum%esum
            osum = temp
        return osum

File: leetcode/easy/3658-gcd-of-odd-and-even-sums/solution.java
class Solution {
    public int gcdOfOddEvenSums(int n) {
        return n*gcd(n, n+1);
        
    }
    public static int gcd(int a, int b){
        while(b != 0){
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
    }
}